app/src/main/java/ar/edu/itba/pam/travelapp/EditTripActivity.java

```diff
@@ -0,0 +1,238 @@
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+                // do nothing :)
+            }
+        });
+        transportSpinner.setSelection(trip.getTravelMethod().ordinal());
+    }
+
+    @Nullable
+    @Override
+    public Object onRetainCustomNonConfigurationInstance() {
+        return presenter;
+    }
+
+
+    @Override
+    public void showDateDialog(EditText inputView) {
+        LocalDate shownDate = DateUtils.parseDate(inputView);
+        if (shownDate == null) {
+            shownDate = LocalDate.now();
+        }
+
+        DatePickerDialog.OnDateSetListener dateSetListener = (view, year, month, dayOfMonth) -> {
+            LocalDate chosenDate = LocalDate.of(year, month + 1, dayOfMonth);
+            inputView.setText(chosenDate.format(dateFormatter));
+        };
+
+        new DatePickerDialog(this, dateSetListener,
+                shownDate.getYear(),
+                shownDate.getMonthValue() - 1,
+                shownDate.getDayOfMonth())
+                .show();
+    }
+
+    @Override
+    public void showDateTimeDialog(EditText inputView) {
+        LocalDateTime shownDateTime = DateUtils.parseDateTime(inputView);
+        if (shownDateTime == null) {
+            shownDateTime = LocalDateTime.now();
+        }
+        dateTimeBuilder = new StringBuilder();
+
+        LocalDateTime finalShownDateTime = shownDateTime;
+        DatePickerDialog.OnDateSetListener dateSetListener = (view, year, month, dayOfMonth) -> {
+
+            LocalDate chosenDate = LocalDate.of(year, month + 1, dayOfMonth);
+            String dateString = chosenDate.format(dateFormatter);
+            dateTimeBuilder.append(dateString).append(" ");
+
+            TimePickerDialog.OnTimeSetListener timeSetListener = (timePicker, hour, minute) -> {
+                LocalDateTime chosenTime = LocalDateTime.of(finalShownDateTime.getYear(), finalShownDateTime.getMonthValue(), finalShownDateTime.getDayOfMonth(), hour, minute);
+                String timeString = chosenTime.format(timeFormatter);
+                dateTimeBuilder.append(timeString);
+                inputView.setText(dateTimeBuilder.toString());
+            };
+
+            new TimePickerDialog(this, timeSetListener,
+                    finalShownDateTime.getHour(),
+                    finalShownDateTime.getMinute(),
+                    false).show();
+        };
+        new DatePickerDialog(this, dateSetListener,
+                shownDateTime.getYear(),
+                shownDateTime.getMonthValue() - 1,
+                shownDateTime.getDayOfMonth()).show();
+    }
+
+
+    @Override
+    public void setErrorMessage(EditText field, String message) {
+        field.setError(message);
+    }
+
+    @Override
+    public void launchDetailsActivity(Trip trip) {
+        Intent intent = new Intent(this, DetailsActivity.class);
+        intent.putExtra("trip", trip);
```

Comment:
Ponelo así para que saque la última activity del stack y no vuelva a la pantalla de edit cuando haces back:

        Intent intent = new Intent(this, DetailsActivity.class);
        intent.putExtra("trip", this.trip);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);